includes/course-theme/class-sensei-course-theme-lesson.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * File containing Sensei_Course_Theme_Lesson class.
+ *
+ * @package sensei-lms
+ * @since 3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Course_Theme_Lesson class.
+ *
+ * @since 3.15.0
+ */
+class Sensei_Course_Theme_Lesson {
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Course_Theme_Lesson constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		if ( 'lesson' !== get_post_type() ) {
+			return;
+		}
+
+		$this->maybe_add_lesson_quiz_results_notice();
+	}
+
+	/**
+	 * Maybe add lesson quiz results notice.
+	 */
+	private function maybe_add_lesson_quiz_results_notice() {
+		$lesson_id = get_the_ID();
+		$user_id   = wp_get_current_user()->ID;
+
+		if ( empty( $lesson_id ) || empty( $user_id ) ) {
+			return '';
+		}
+
+		$quiz_permalink = Sensei()->lesson->get_quiz_permalink( $lesson_id );
+
+		if ( empty( $quiz_permalink ) ) {
+			return '';
+		}
+
+		if ( ! Sensei()->lesson->is_quiz_submitted( $lesson_id, $user_id ) ) {
+			return '';
```

Comment:
No need to return an empty string, right? You can just `return;` so it is consistent in terms of output when actually doing something below.

Reply:
That's correct! This part started as a block render and after changing it, I missed that!
Fixed here: https://github.com/Automattic/sensei/pull/4477/commits/2e51c4a7ed820650110658b2f3272919adf4c897